### PR TITLE
Clang format 14

### PR DESCRIPTION
**Pull request by rscampos**

### CODE REVIEW

File: libbpfgo.c [hunk 1]
```diff
@@ -6,8 +6,8 @@ extern void perfLostCallback(void *ctx, int cpu, __u64 cnt);
 extern int ringbufferCallback(void *ctx, void *data, size_t size);
 
 int libbpf_print_fn(enum libbpf_print_level level, // libbpf print level
-                    const char *format,            // format used for the msg
-                    va_list args)                  // args used by format
+                    const char *format, // format used for the msg
```
Comment (geyslan) [inline libbpfgo.c]: I do prefer the old indentation, it resembles Golang formatting. Which clang-format option should be returned?

Comment (geyslan) [reply libbpfgo.c]: I think it's `AlignTrailingComments`.

File: libbpfgo.c [hunk 2]
```diff
@@ -121,13 +126,10 @@ int cgo_bpf_prog_detach_cgroup_legacy(int prog_fd,   // eBPF program file descri
...
 
-struct bpf_iter_attach_opts *cgo_bpf_iter_attach_opts_new(__u32 map_fd,
-                                                          enum bpf_cgroup_iter_order order,
-                                                          __u32 cgroup_fd,
-                                                          __u64 cgroup_id,
-                                                          __u32 tid,
-                                                          __u32 pid,
-                                                          __u32 pid_fd)
+struct bpf_iter_attach_opts *
+cgo_bpf_iter_attach_opts_new(__u32 map_fd, enum bpf_cgroup_iter_order order,
```
Comment (geyslan) [inline libbpfgo.c]: You can notice that some wrapping is being done due to the limit `PenaltyReturnTypeOnItsOwnLine`. So some functions with the small signature has the return type and name in the same line and others will have it with break line.

That's my proposal, enforce it as FreeBSD does: https://github.com/freebsd/freebsd-src/blob/main/.clang-format.

Can we give it a try, disregarding the linux one?

File: libbpfgo.c [hunk 3]
```diff
@@ -189,15 +191,10 @@ void cgo_bpf_object_open_opts_free(struct bpf_object_open_opts *opts)
...
-                                                        __u32 btf_vmlinux_value_type_id,
-                                                        __u32 inner_map_fd,
-                                                        __u32 map_flags,
-                                                        __u64 map_extra,
-                                                        __u32 numa_node,
-                                                        __u32 map_ifindex)
+struct bpf_map_create_opts *cgo_bpf_map_create_opts_new(
+    __u32 btf_fd, __u32 btf_key_type_id, __u32 btf_value_type_id,
+    __u32 btf_vmlinux_value_type_id, __u32 inner_map_fd, __u32 map_flags,
+    __u64 map_extra, __u32 numa_node, __u32 map_ifindex)
```
Comment (geyslan) [inline libbpfgo.c]: That's other oddity (ugliness)... some functions will align its params using the first line, others like this will break to the next.

**Comment by rscampos** [comment]
I've pushed a modified PR, without touching `.clang_format`, in future we can work on that together. Agreed with the suggestions.

**Pull request merged**